### PR TITLE
Add website edit form to lookup tool (bug 1161867)

### PR DESCRIPTION
**Pull request by robhudson**

### CODE REVIEW

**Comment by robhudson** [comment]
Ignore the black bar... the built-in developer toolbar screenshot keeps adding that for some reason.
screen shot 2015-06-19 at 12 51 47

File: mkt/lookup/views.py [hunk 1]
```diff
@@ -39,19 +39,19 @@
...
 from mkt.webapps.models import Webapp
 from mkt.websites.models import Website
+from mkt.websites.forms import WebsiteForm
 from mkt.websites.views import WebsiteSearchView
 
 
 log = commonware.log.getLogger('z.lookup')
 
 
-@login_required
```
Comment (eviljeff) [inline mkt/lookup/views.py]: not needed when permissions are asked for?

Comment (robhudson) [inline mkt/lookup/views.py]: Right, `@permission_required` does `@login_required` already.

File: mkt/lookup/views.py [hunk 2]
```diff
@@ -329,7 +323,23 @@ def website_summary(request, addon_id):
     })
 
 
-@login_required
+@permission_required([('WebsiteLookup', 'View')])
+def website_edit(request, addon_id):
```
Comment (eviljeff) [inline mkt/lookup/views.py]: addon_id :cry:

Comment (robhudson) [inline mkt/lookup/views.py]: heh, yeah. I guess I could change it. But everything else passes around `addon_id`s.

File: mkt/websites/forms.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            regions = map(int, self.cleaned_data.get('preferred_regions'))
+        except (TypeError, ValueError):
+            # Data is not a list or data contains non-integers.
+            raise forms.ValidationError(_('Invalid region(s) selected.'))
+
+        return list(regions)
+
+    def save(self, commit=False):
+        form = super(WebsiteForm, self).save(commit=False)
+        form.save()
```
Comment (eviljeff) [inline mkt/websites/forms.py]: this looks... weird.

File: mkt/lookup/tests/test_views.py [hunk 4]
```diff
@@ -1211,3 +1212,22 @@ def test_by_name(self):
     def test_by_id(self):
         data = self.search(q=self.website.pk)
         self.verify_result(data)
+
+
+class TestWebsiteEdit(mkt.site.tests.TestCase):
+    fixtures = fixture('user_support_staff', 'user_999')
```
Comment (eviljeff) [inline mkt/lookup/tests/test_views.py]: user_999 seems unused.

**Comment by eviljeff** [comment]
r+wc but I'm fairly unfamiliar with the MOW implementation so far.

File: mkt/websites/forms.py (hunk 3 above)
Comment (robhudson) [inline mkt/websites/forms.py]: Hmm, yes it does. I'll double check this, thanks.

File: mkt/lookup/tests/test_views.py [hunk 5]
```diff
@@ -1211,3 +1212,22 @@ def test_by_name(self):
     def test_by_id(self):
         data = self.search(q=self.website.pk)
         self.verify_result(data)
+
+
+class TestWebsiteEdit(mkt.site.tests.TestCase):
```
Comment (diox) [inline mkt/lookup/tests/test_views.py]: Could/should we have more view tests ?

Comment (robhudson) [inline mkt/lookup/tests/test_views.py]: The view is actually pretty empty, most of the logic is in the form, which is also where the tests are. But if you have something specific I'd be happy to add it. Perhaps just a simple test that a form submission redirects and updates the data as a view test?

Comment (diox) [inline mkt/lookup/tests/test_views.py]: Yeah, simple test that loads and then posts the form should be enough

**Comment by diox** [comment]
r+wc once the view tests pass

**Comment by robhudson** [comment]
Ha, yep. I'm rebuilding my docker to run the tests locally. I'll merge once the test is good. Thanks for the review.

File: mkt/websites/forms.py [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+                               widget=forms.Textarea(attrs={'rows': 2}))
+    name = TransField(label=_lazy(u'Name'), widget=TransInput())
+    preferred_regions = forms.MultipleChoiceField(
+        label=_lazy(u'Preferred Regions'), choices=REGIONS_CHOICES_NAME,
+        required=False, widget=forms.SelectMultiple(attrs={'size': 10}))
+    short_name = TransField(label=_lazy(u'Short Name'), widget=TransInput(),
+                            required=False)
+    title = TransField(label=_lazy(u'Title'), widget=TransInput(),
+                       required=False)
+    url = forms.URLField(label=_lazy(u'URL'))
```
Comment (diox) [inline mkt/websites/forms.py]: Are you having this field just because you want it to be required ? Maybe I should not have allowed blanks in the model...

Comment (robhudson) [inline mkt/websites/forms.py]: If it's definitely required then yeah, we probably want to not allow it to be empty at the model level too. Should we do that here or in a follow up commit?

**Comment by robhudson** [comment]
https://github.com/mozilla/zamboni/commit/e3818e9

**Pull request closed without merging**